src/lib/features/segment/segment-read-model.ts

```diff
@@ -111,7 +111,7 @@ export class SegmentReadModel implements ISegmentReadModel {
     }
 
     async getAllForClientIds(ids?: number[]): Promise<IClientSegment[]> {
-        if (!ids || ids.length === 0) {
+        if (ids?.length === 0) {
```

Comment:
Is this actually doing the thing?

In the left side it was returning [] if id was undefined or null or if length was zero. In the right side, it looks like the same...

Reply:
Ok, in this case if ids is undefined it will provide all the segments, while it only returns empty if the ids provided is the empty array.

Reply:
Yep, messed with my brain a bit as well. Essentially the happy path is `ids` falsy but not an empty array, the highlighted part on my screenshot (no "early return" is logged).

![image](https://github.com/user-attachments/assets/d15de70e-aefc-4be9-8743-cc02886f6be1)

Reply:
Yes, correct